src/toBeSorted/blur_and_palette_manager.cpp

```diff
@@ -1,12 +1,266 @@
 #include "toBeSorted/blur_and_palette_manager.h"
 
+#include "common.h"
+#include "d/a/d_a_player.h"
 #include "m/m_color.h"
+#include "m/m_vec.h"
+#include "nw4r/math/math_arithmetic.h"
 
 BlurAndPaletteManager BlurAndPaletteManager::sInstance;
 BlurAndPaletteManager *BlurAndPaletteManager::sPInstance;
 
 TList<UnkBlurPaletteListNode, offsetof(UnkBlurPaletteListNode, mNode)> sPlayingEffectsList;
 
+// Same as vectle_calc in d_kankyo_rain from tp
+static void vectle_calc(const mVec3_c *pIn, mVec3_c *pOut) {
+    f32 s = nw4r::math::FSqrt(pIn->squaredLength());
+    if (s != 0.0f) {
+        pOut->x = pIn->x / s;
+        pOut->y = pIn->y / s;
+        pOut->z = pIn->z / s;
+    } else {
+        pOut->x = 0.0f;
+        pOut->y = 0.0f;
+        pOut->z = 0.0f;
+    }
+}
+
+// Same as get_vectle_calc in d_kankyo_rain from tp
+void BlurAndPaletteManager::get_vectle_calc(const mVec3_c *pInA, const mVec3_c *pInB, mVec3_c *pOut) {
+    mVec3_c pos;
+    pos.x = pInB->x - pInA->x;
+    pos.y = pInB->y - pInA->y;
+    pos.z = pInB->z - pInA->z;
+    vectle_calc(&pos, pOut);
+}
+
+// TODO(??)
+inline f32 sinf(f32 x) {
+    return sin(x);
+}
+
+// TODO(??)
+inline f32 cosf(f32 x) {
+    return cos(x);
+}
```

Comment:
These are probably the same `std::cosf` / `std::sinf` I had to add for SND, hence my suggestion that we work on our MSL_C headers at some point.